src/main/java/org/wildfly/security/auth/provider/SimpleMapBackedSecurityRealm.java

```diff
@@ -118,5 +121,17 @@ public CredentialSupport getCredentialSupport(Class<?> credentialType) {
             final Password password = map.get(principal);
             return credentialType.isInstance(password) ? credentialType.cast(password) : null;
         }
+
+        public VerificationResult verifyCredential(final Object credential) {
```

Comment:
There is only going to be a very limited subset of credential types that would be usable with this API think, just thinking do we need some form of getCredentialSupport or enhancement of getCredentialSupport to identify which ones are supported for verification.